person.rb

```diff
@@ -1,17 +1,18 @@
 require 'nameable'
 
 class Person < Nameable
+  attr_reader :id
+  attr_accessor :name, :age, :rentals
+
   def initialize(age, name = 'Unknown', parent_permission: true)
     super(age, name, parent_permission: parent_permission)
```

Comment:
- [ ] [Optional]👍 Great job on addressing the previous reviewer's feedback and making improvements. However, it seems there's an issue when trying to create a new person. you shouldn't send user details to the superclass when creating a person. The `Nameable` class is the base Component interface and defines operations that can be altered by decorators. You don't need to send user details to the `Nameable` class.

![image](https://github.com/zhorabay/School-library/assets/55840999/518f8cb7-9711-449b-8aab-a823d59acd95)

By making this adjustment, your code should work as expected, and you can create a new person without any errors. Keep up the great work! 👍👨‍💻

Reply:
Hello,
Thank you for your review. As I understand in order to fix this issue I should remove 'Nameable' from line 3, right? and in this case I should not require nameable.rb, too. I also face this issue in my ends, but also it seems it is not correct to remove it from inherits.

Reply:
@zhorabay, Hi, I did not request to remove 'Nameable' as it is one of the project requirements. But I notice that you are currently using `require 'nameable'`. you also have to use `require_relative 'nameable'`. Hope this will fix your errors.